src/Plugin.php

```diff
@@ -11,22 +11,222 @@
 
 use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
 use Composer\IO\IOInterface;
+use Composer\Package\AliasPackage;
+use Composer\Package\PackageInterface;
 use Composer\Plugin\PluginInterface;
+use Composer\Script\ScriptEvents;
+use Symfony\Component\Process\Exception\LogicException;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Symfony\Component\Process\Process;
 
 /**
  * PHP_CodeSniffer standard installation manager.
  *
  * @author Franck Nijhof <f.nijhof@dealerdirect.nl>
  */
-class Plugin implements PluginInterface
+class Plugin implements PluginInterface, EventSubscriberInterface
 {
+
+    const PACKAGE_TYPE = 'phpcodesniffer-standard';
+
+    /**
+     * @var Composer
+     */
+    protected $composer;
+
+    /**
+     * @var IOInterface
+     */
+    protected $io;
+
+    /**
+     * @var array
+     */
+    private $installedPaths;
+
+    /**
+     * @var string
+     */
+    private $phpCodeSnifferBin;
+
     /**
      * {@inheritDoc}
+     *
+     * @throws \RuntimeException
+     * @throws LogicException
+     * @throws RuntimeException
+     * @throws ProcessFailedException
      */
     public function activate(Composer $composer, IOInterface $io)
     {
-        $installer = new Installer($io, $composer);
-        $composer->getInstallationManager()->addInstaller($installer);
+        $this->composer = $composer;
+        $this->io = $io;
+        $this->installedPaths = [];
+        $this->phpCodeSnifferBin = $composer->getConfig()->get('bin-dir') . DIRECTORY_SEPARATOR . 'phpcs';
+
+        $this->loadInstalledPaths();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ScriptEvents::POST_INSTALL_CMD => [
+                ['onScriptPost', 0],
+            ],
+            ScriptEvents::POST_UPDATE_CMD => [
+                ['onScriptPost', 0],
+            ],
+        ];
+    }
+
+    /**
+     * Entry point for post install and post update events
+     *
+     * @throws RuntimeException
+     * @throws LogicException
+     * @throws ProcessFailedException
+     */
+    public function onScriptPost()
+    {
+        // Ensure PHP_CodeSniffer is installed
+        if ($this->isPHPCodeSnifferInstalled() === false) {
+            return;
+        }
+
+        // Clean and update. Trigger a save when something changed.
+        if ($this->cleanInstalledPaths() === true || $this->updateInstalledPaths() === true) {
+            $this->saveInstalledPaths();
+        }
+    }
+
+    /**
+     * Load all paths from PHP_CodeSniffer into an array
+     *
+     * @throws RuntimeException
+     * @throws LogicException
+     * @throws ProcessFailedException
+     */
+    private function loadInstalledPaths()
+    {
+        $phpcs = new Process($this->phpCodeSnifferBin . ' --config-show installed_paths');
```

Comment:
Instead of creating a new `Process` three times you might want to look into using the ProcessBuilder factory.

```
$this->builder = new ProcessBuilder();
$this->builder->setPrefix($this->phpCodeSnifferBin);
...
$this->builder
    ->setArguments(array('--list', '--file=archive.tar.gz'))
    ->getProcess()
    ->mustRun() /* or ->run()*/
```